Simplenote/src/main/java/com/automattic/simplenote/di/ThreadModule.kt

```diff
@@ -0,0 +1,23 @@
+package com.automattic.simplenote.di
+
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import javax.inject.Named
+
+const val IO_THREAD = "IO_THREAD"
+
+@Module
+@InstallIn(SingletonComponent::class)
+abstract class ThreadModule {
```

Comment:
This is because `runBlockingTest` still doesn't manage multithreading well. There is an [open issue](https://github.com/Kotlin/kotlinx.coroutines/issues/1222) that discusses something about it. They try a fix but was not successful and found that the change required is more complex. 

> This is totally unrelated to this PR, but SimperiumTagsRepository is untested, maybe we should create an issue to make sure that to cover this repository with adequate tests as well. Wdty?

Yes, definitely. I will add tests to this repo in this sprint.

Reply:
> This is because runBlockingTest still doesn't manage multithreading well. There is an open issue that discusses something about it. They tried a fix but was not successful and found that the change required is more complex.

Hmmm... this is very interesting. So, the problem is purely because of testing, right? If so, what kind of problems did you find while doing testing without the `ThreadModule` alternative?

Questions (❓): If we are to stick with the `ThreadModule`, should the `SimperiumTagsRepository` be updated as well?

Reply:
>  If so, what kind of problems did you find while doing testing without the ThreadModule alternative?

Mostly `IllegalStateException: This job has not completed yet` exceptions. Even though I was running the suspend functions inside blocks of `runBlockingTest`, the coroutines did not finish executing before I assert the results.

Reply:
> If we are to stick with the ThreadModule, should the SimperiumTagsRepository be updated as well?

Yes. Fixed in 32eca5b829a1a552d1bbb82ad0144a4393e91e2d.

Reply:
👋 @danilo04 !

> Mostly IllegalStateException: This job has not completed yet exceptions. Even though I was running the suspend functions inside blocks of runBlockingTest, the coroutines did not finish executing before I assert the results.

I see... thanks once more for the elaborate explanation and for updating `SimperiumTagsRepository` as well! 🚀